### PR TITLE
USWDS - Links: Update external link guidance

### PR DESCRIPTION
**Pull request by mejiaj**
# Summary

Updated link guidance and changelog.

- Created `_data/settings/link.yml` for new settings
- Updated changelog with two entries

## Related issue

Guidance portion of uswds/uswds#5164.

## Preview link

<details><summary>Component guidance</summary>

Entries for changelog and documenting component settings. [Preview link →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-external-link-sr-guidance/components/link/)

image

image
</details> 

<details><summary>Settings table in Documentation</summary>

New section for link component in settings table. [Preview link →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-external-link-sr-guidance/documentation/settings/#link-2)

### Table
image

### Changelog
image

</details>

## Major changes

- Added `noreferrer` note to guidance
- Updated link changelog

## Testing and review

_Share recommended methods for reviewing this change._

### CODE REVIEW

**Comment by amyleadem** [comment]
Hi @mejiaj, could this PR also add `$theme-external-link-sr-label-tab-new` and `$theme-external-link-sr-label-tab-same` to the settings page (with a changelog update)?

**Comment by mejiaj** [comment]
@amyleadem I'll work on making these changes today.

**Review by mahoneycm — APPROVED** [review]
Looking good!

- [x] Guidance is clear and effective
- [x] External link sr settings match USWDS settings and are concise and descriptive
- [x] Change logs look good

File: _components/link/guidance/usability-should.md [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 - **If you use an external link indicator, use it consistently for all text links.** If your project uses an external link indicator (like an icon), use it for all text links across your site. If users learn to associate an external link with the indicator, they will also expect that text links without an indicator are not external links. Icon- or image-only links like social share buttons or logos do not need the same treatment as text links.
 
+- **Use `rel="noreferrer"` property on external links.** Setting `noreferrer` on links prevents the browser leaking referrer information.
```
Comment (amyleadem) [inline _components/link/guidance/usability-should.md:17]: Optional:
This works but I had some questions and I wonder if users might also have them:
Do we recommend this on every external link? Is there a time when users might want to exclude this? Will users understand what "leaking referrer information" means?

File: _data/settings/components/link.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: Link
+contents:
+  - name: New tab screen reader label
+    description: Screen reader label for external links in new tabs.
```
Comment (amyleadem) [inline _data/settings/components/link.yml:4]:
```suggestion
    description: Screen reader label for external links that open in new tabs.
```

File: _data/settings/components/link.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    var: $theme-external-link-sr-label-tab-new
+    default: '"External, opens in a new tab."'
+    type: string
+  - name: Same tab screen reader label
+    description: Screen reader label for external links in same tab.
```
Comment (amyleadem) [inline _data/settings/components/link.yml:9]:
```suggestion
    description: Screen reader label for external links that open in current tab.
```

**Review by amyleadem — COMMENTED** [review]
Looks good. I added a couple of comments to see if we can make things a bit clearer, but they are not blocking changes.

File: _components/link/guidance/usability-should.md (hunk 1 above)
Comment (thisisdano) [reply _components/link/guidance/usability-should.md:17]: Re: "leaking referrer information" — I'd say no. This needs a bit more plain-language translation

Comment (mejiaj) [reply _components/link/guidance/usability-should.md]: Good points, updated in 43fe347. It now reads:

>Setting `noreferrer` on links prevents the browser leaking information about the original web address.

_Based on this [MDN guidance on referer](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer#:~:text=When%20you%20click%20a%20link%2C%20the%20Referer%20contains%20the%20address%20of%20the%20page%20that%20includes%20the%20link.%20When%20you%20make%20resource%20requests%20to%20another%20domain%2C%20the%20Referer%20contains%20the%20address%20of%20the%20page%20that%20uses%20the%20requested%20resource.)._

**Review by thisisdano — APPROVED** [review]

**Pull request merged**